triedb/pathdb/history_indexer.go

```diff
@@ -507,6 +507,8 @@ func (i *indexIniter) index(done chan struct{}, interrupt *atomic.Int32, lastID
 			storeIndexMetadata(i.disk, 0)
 			log.Info("Initialized history indexing flag")
 		} else {
+			// Update the metadata to indicate that indexing is complete up to lastID
```

Comment:
I think we should error out if `beginID > lastID+1`? It's not expected.

We can't blindly reset the index marker to `lastID`, the root cause should be fixed instead.